File: Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        subStringSet = set()
        longestSubString = 0
        for right in range(len(s)):
            while s[right] in subStringSet:
                subStringSet.remove(s[left])
                left += 1
            subStringSet.add(s[right])
            longestSubString = max(longestSubString, right-left+1)
        return longestSubString

File: Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.cpp
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        set<char> subStringSet;
        int left = 0;
        int longestSubString = 0;
        for (int right = 0; right < s.size(); right++)
        {
            while (subStringSet.find(s[right]) != subStringSet.end())
            {
                subStringSet.erase(s[left]);
                left++;
            }
            subStringSet.insert(s[right]);
            longestSubString = max(longestSubString, right - left + 1);
        }
        return longestSubString;
    }
};